src/schema/jsii2schema.ts

```diff
@@ -445,13 +450,39 @@ export function schemaForEnumLikeClass(
     },
   });
 
-  if (anyOf.length === 0) {
-    return undefined;
-  }
+  anyOf.push({
```

Comment:
Where does this check if the class can be instantiated? (Class can only be instantiated if its constructor is not `private` or `protected`)

Reply:
I totally missed that, thanks! Fixed.